### PR TITLE
fix scoring logic and stop game if score reaches 0

### PR DESCRIPTION
**Pull request by VijayantSoni**

### CODE REVIEW

File: playlist.js [hunk 1]
```diff
@@ -53,6 +58,7 @@ function playGame(metadata, difficulty) {
         } else {
           snackNotif('Challenge Complete', 5000);
           document.getElementById('playlist-ready').innerHTML = 'Challenge Complete!';
+          document.getElementById('start-challenge').disabled = false;
```
Comment (VijayantSoni) [inline playlist.js:60]: see if this makes sense. disabling button until current game is over

File: playlist.js [hunk 2]
```diff
@@ -25,16 +27,19 @@ function shuffle(array) {
...
+
+    current_score_degrade_constant = scoringMetadata.score_degrade_constant
+
     fileArray = shuffle(fileArray);
 
     maxScore = fileArray.length * scoringMetadata.item_score
     document.getElementById('score').innerHTML = maxScore
 
     document.getElementById('start-challenge').disabled = true;
+    document.getElementById('playlist-ready').innerHTML = 'Challenge Playlist Ready!';
```
Comment (ashshdwvdi) [inline playlist.js]: lets have a common method for this like showPlaylistReady ?

File: playlist.js (hunk 1 above)
Comment (ashshdwvdi) [reply playlist.js:60]: it looks good to me

File: script.js [hunk 3]
```diff
@@ -35,26 +34,43 @@ function startVideo() {
...
+  // Next degrade value
+  current_score_degrade_constant = current_score_degrade_constant * scoringMetadata.score_degrade_exponential_factor
+
+  if (new_score <= 0) {
+    snackNotif('Game over !!!', 5000);
+    document.getElementById('playlist-ready').innerHTML = 'Game over !!!'
+    document.getElementById('challenge').pause()
+    document.getElementById('start-challenge').disabled = false;
+    document.getElementById('score').innerHTML = 0
+    return
```
Comment (ashshdwvdi) [inline script.js]: Create a new method and put it there like `displayGameOver` helpful later.

**Review by mrgreen-creates — APPROVED** [review]

**Comment by mrgreen-creates** [comment]
if its okay can we merge this first v2.0 https://github.com/harshmandalgi/laugh-challenge/pull/15
and then you can update your branch once and also use `showNotificationAlert` when the player wins/loses instead of `snackNotif`

**Pull request merged**